### ISSUE TITLE
Exposing gcalcli as an API to other programming languages (maybe using JSON)

### ISSUE THREAD

**Issue by aindilis** [insanum/gcalcli#466]
I have a large system written in Perl / Prolog that I'm trying to sync with GCal, and I was unable to get the GCal clients for Perl to work, when I came across gcalcli, which worked great!  Thank you!  I am wondering is there a way to programmatically access gcalcli, such as via a RESTful API, or via a wrapper, or something.  I would like to avoid parsing the STDOUT of the gcalcli CLI program, but if I have to, are there ways to have it dump the entire event spec, hopefully in JSON, etc.  I know very little Python.

**Comment by jcrowgey** [comment]
For now, there's no json output option built into the program.  Depending on what information you're trying to extract, the --tsv option may give you something close enough.  Alternatively, the `json` package is in the Python standard library, so hacking `json.dumps(event)` into the code somewhere shouldn't be hard.

I'd be interested in a json output PR if you make something nice.

**Comment by shaicoleman** [comment]
You'd might want to use [google-calendar3-cli](https://github.com/Byron/google-apis-rs/tree/main/gen/calendar3-cli) for that purpose

**Comment by dbarnett** [comment]
Yeah, if anyone has use cases for things they'd like to do programmatically like this from python, I could take a look and think over options. IMO it's at least worth documenting any good related libraries if not exposing our own.

Does anyone know if https://github.com/Byron/google-apis-rs/tree/main/gen/calendar3-cli is easy to use as a dep of a python project or in other cases where someone might want programmatic APIs? From a glance I'd guess you'd use it by invoking an executable, not importing a library, but that may be a fine approach depending on what someone's wanting to do with it.
